### ISSUE TITLE
Incorrect parsing of branches?

### ISSUE THREAD

**Issue by timholy** [tonyhffong/Lint.jl#3]
If you define a file whose contents are

``` julia
wrap(pos::Int, len::Int) = pos < 1 ? 1 : (pos > len ? len : pos)
```

then Lint gives:

```
julia> lintfile("/tmp/wrap.jl")
        /tmp/wrap.jl [                wrap]    1 WARN  false branch is unreachable
```

but

```
julia> include("/tmp/wrap.jl")
wrap (generic function with 1 method)

julia> wrap(7,10)
7

julia> wrap(-3,10)
1

julia> wrap(15,10)
10
```
